dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePlugin.kt

```diff
@@ -43,17 +47,32 @@ class DdAndroidGradlePlugin @Inject constructor(
         // need to use withPlugin instead of afterEvaluate, because otherwise generated assets
         // folder with buildId is not picked by AGP by some reason
         target.pluginManager.withPlugin("com.android.application") {
-            val androidExtension = target.androidApplicationExtension ?: return@withPlugin
-            androidExtension.applicationVariants.all { variant ->
-                if (extension.enabled) {
+            // can work probably even with lower versions, but legacy Variant API is working fine there as well
+            @Suppress("MagicNumber")
+            if (TaskUtils.isAgpAbove(8, 4, 0) &&
```

Comment:
> [!NOTE]
> Minor: it would probably be best to have this hardcoded version in a Single Source of Truth String, and have extracted static fields. E.g.:

```
companion object {
    const val AGP_VARIANT_API = "8.4.0"
    val AGP_VARIANT_API_MAJ = AGP_VARIANT_API.split('.')[0].toInt()
    val AGP_VARIANT_API_MIN = AGP_VARIANT_API.split('.')[1].toInt()
    val AGP_VARIANT_API_FIX = AGP_VARIANT_API.split('.')[2].toInt()
}
```

Reply:
@xgouchet I created an API like this https://github.com/DataDog/dd-sdk-android-gradle-plugin/pull/263/files#diff-881fec92f3bb5fd22766703d87e3df1380ed61a0ec5ca454fd4c4b97e0759e91 instead. Let me know what you think.